datafusion/physical-plan/src/test.rs

```diff
@@ -0,0 +1,343 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Utilities for testing datafusion-physical-plan
+
+use std::error::Error;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::{collections::HashMap, path::PathBuf};
+
+use arrow_array::{ArrayRef, Int32Array, RecordBatch};
+use arrow_schema::{DataType, Field, Schema, SchemaRef};
+use futures::{Future, FutureExt};
+
+use crate::memory::MemoryExec;
+use crate::ExecutionPlan;
+
+pub mod exec;
+
+/// A macro to assert that one string is contained within another with
```

Comment:
These functions appear to now be duplicated?

Perhaps some of these could be moved to datafusion-common::test_util? Perhaps with a `#[doc(hidden)]` if we want to hide them from the public API

Reply:
I tried to move the macros but I couldn't figure out how to get them to be exported correctly. I'll give it another go

Reply:
Thank you -- this was an excellent suggestion. I was able to move all the testing macros to `datafusion_common` and remove the copies in core and physical-plan